ui/src/components/SampleView/ContextMenu.jsx

```diff
@@ -424,24 +423,20 @@ export default class ContextMenu extends React.Component {
     this.props.sampleViewActions.deleteShape(this.props.shape.id);
   }
 
-  measureDistance() {
-    this.props.sampleViewActions.measureDistance(true);
-  }
-
-  toggleDrawGrid() {
-    this.props.sampleViewActions.toggleDrawGrid();
-  }
-
   saveGrid() {
-    const gd = { ...this.props.shape.gridData };
-    this.props.sampleViewActions.addShape({ t: 'G', ...gd });
+    const { gridData } = this.props.shape;
+    this.props.sampleViewActions.addShape({ t: 'G', ...gridData });
     this.props.sampleViewActions.toggleDrawGrid();
   }
 
   createPointAndShowModal(name, extraParams = {}) {
-    const { imageX, imageY, imageRatio } = this.props;
-    this.createPoint(imageX / imageRatio, imageY / imageRatio, (shape) =>
-      this.showModal(name, {}, shape, extraParams),
+    const { sampleViewX, sampleViewY } = this.props;
+
+    this.props.sampleViewActions.add2DPoint(
+      sampleViewX,
+      sampleViewY,
+      'SAVED',
+      (shape) => this.showModal(name, {}, shape, extraParams),
```

Comment:
Maybe you already thought about it and want to do that in a different PR, but related to #1490: Couldn't we just prevent the drawing of a point when no sample is mounted by adding a check for `sampleData` (similar to how it is done in `showModal`)

Reply:
You are correct, that might be a more elegant solution to the problem. Sorry, It seems that I misunderstood the other comment, the first time I read it